tests/View/Blade/BladeCustomTest.php

```diff
@@ -50,6 +50,34 @@ public function testCustomShortStatements()
         $this->assertEquals($expected, $this->compiler->compileString($string));
     }
 
+    public function testValidCustomNames()
+    {
+        $this->assertNull($this->compiler->directive('custom', function () {
+        }));
+        $this->assertNull($this->compiler->directive('custom_custom', function () {
+        }));
+        $this->assertNull($this->compiler->directive('customCustom', function () {
+        }));
+        $this->assertNull($this->compiler->directive('custom::custom', function () {
+        }));
+    }
+
+    public function testInvalidCustomNames()
```

Comment:
maybe. will be better to use dataPrrovider?

Reply:
you'll have to explain this more. I don't know what you mean.

Reply:
You can find more information about data providers at https://phpunit.de/manual/6.5/en/writing-tests-for-phpunit.html

Reply:
ahh, got it.  thanks @sisve. I think what I have is fine, don't see any compelling reason to use a data provider.